### PR TITLE
Mirantis MKE Engine

### LINKED ISSUE #1181: Mirantis K8s engine support
## Feature Request

**Short Description**

Supporting [k8s cluster solution from Mirantis](https://www.mirantis.com/download/mirantis-cloud-native-platform/mirantis-kubernetes-engine/).

**Describe the solution you'd like**

- [ ] Check if policy enforcement, observability, audit rules, network segmentation is supported
- [ ] Get output of `karmor probe`
- [ ] Update [kubearmor support matrix](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/support_matrix.md)
- [ ] Provide and attach `karmor sysdump`

### PR DESCRIPTION
**Pull request by AshokAccuknox**
KubeArmor is supporting Mirantis Kubernetes Engine (MKE).

I have tested with MKE running in Bare metal environment. KubeArmor policies are enforced.

KubeArmor is Supporting Mirantis Kubernetes Engine (MKE).

Please check the document attached with the issue  #1181

Attaching

[karmor-sysdump-Sun Mar 26 10_46_23 IST 2023.zip](https://github.com/kubearmor/KubeArmor/files/11070824/karmor-sysdump-Sun.Mar.26.10_46_23.IST.2023.zip)
[karmor probe.txt](https://github.com/kubearmor/KubeArmor/files/11070829/karmor.probe.txt)

**Purpose of PR?**:

Fixes #1181

**Does this PR introduce a breaking change?**

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [x] Bug fix. Fixes #1181
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`

### CODE REVIEW

**Comment by nyrahul** [comment]
@AshokAccuknox , the doc referenced is your org's internal doc. Secondly, please add DCO to the commits.

File: getting-started/support_matrix.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ KubeArmor supports following types of workloads:
 | Rancher    | [K3S] | [Distros] | x86_64 | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | [BPFLSM], AppArmor |
 | Oracle     | [Ampere] | [UEK] | ARM | :heavy_check_mark: | :heavy_check_mark: | :x: | :heavy_check_mark: | :heavy_check_mark: | [1084] |
 | VMWare     | [Tanzu] | TBD | x86_64 | :construction: | :construction: | :construction: | :construction: | :construction: | [1064] |
+| Mirantis     | [k8s] | Ubuntu>- 20.04 | x86_64 | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | [1181] |
```
Comment (Ankurk99) [inline getting-started/support_matrix.md]: Please mention that it's tested with AppArmor

**Comment by Ankurk99** [comment]
> the doc referenced is your org's internal doc.

Can we move this to KubeArmor wiki?

File: getting-started/support_matrix.md (hunk 1 above)
Comment (AshokAccuknox) [reply getting-started/support_matrix.md]: Added the Document in the Issues and also completed the changes that Ankur mentioned it.

**Comment by AshokAccuknox** [comment]
> @AshokAccuknox , the doc referenced is your org's internal doc. Secondly, please add DCO to the commits.

I have attached the document in Issues and also added DCO in Commits.

**Review by nyrahul — APPROVED** [review]

File: getting-started/support_matrix.md [hunk 2]
```diff
@@ -29,6 +29,7 @@ KubeArmor supports following types of workloads:
 | Rancher    | [K3S] | [Distros] | x86_64 | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | [BPFLSM], AppArmor |
 | Oracle     | [Ampere] | [UEK] | ARM | :heavy_check_mark: | :heavy_check_mark: | :x: | :heavy_check_mark: | :heavy_check_mark: | [1084] |
 | VMWare     | [Tanzu] | TBD | x86_64 | :construction: | :construction: | :construction: | :construction: | :construction: | [1064] |
+| Mirantis     | [k8s] | Ubuntu>- 20.04 | x86_64 | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | AppArmor | [1181] |
```
Comment (nyrahul) [inline getting-started/support_matrix.md]:
```suggestion
| Mirantis     | [MKE] | Ubuntu>=20.04 | x86_64 | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | AppArmor | [1181] |
```

Comment (nyrahul) [reply getting-started/support_matrix.md]: You need to add [MKE] and [1181]  refs below.

**Review by nyrahul — CHANGES_REQUESTED** [review]

File: getting-started/support_matrix.md (hunk 2 above)
Comment (AshokAccuknox) [reply getting-started/support_matrix.md]: I have made the necessary changes.

**Comment by Ankurk99** [comment]
@AshokAccuknox Can you please squash all the commits into one?

**Comment by AshokAccuknox** [comment]
> @AshokAccuknox Can you please squash all the commits into one?
Squashed all commits into one commit

**Review by Ankurk99 — APPROVED** [review]
LGTM!

File: getting-started/support_matrix.md [hunk 3]
```diff
@@ -29,6 +29,7 @@ KubeArmor supports following types of workloads:
 | Rancher    | [K3S] | [Distros] | x86_64 | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | [BPFLSM], AppArmor |
 | Oracle     | [Ampere] | [UEK] | ARM | :heavy_check_mark: | :heavy_check_mark: | :x: | :heavy_check_mark: | :heavy_check_mark: | [1084] |
```
Comment (nyrahul) [inline getting-started/support_matrix.md]:
```suggestion
| Oracle     | [Ampere] | [UEK] | ARM | :heavy_check_mark: | :heavy_check_mark: | :x: | :heavy_check_mark: | SELinux | [1084] |
```

**Review by nyrahul — CHANGES_REQUESTED** [review]
@Ankurk99 can we make a quick fix for Oracle Ampere as suggested?

**Review by nyrahul — APPROVED** [review]
LGTM

**Pull request merged**